admin/server/github.go

```diff
@@ -237,6 +237,92 @@ func (s *Server) GetGithubRepoStatus(ctx context.Context, req *adminv1.GetGithub
 	return res, nil
 }
 
+func (s *Server) ListGithubUserRepos(ctx context.Context, req *adminv1.ListGithubUserReposRequest) (*adminv1.ListGithubUserReposResponse, error) {
+	claims := auth.GetClaims(ctx)
+	if claims.OwnerType() != auth.OwnerTypeUser {
+		return nil, status.Error(codes.Unauthenticated, "not authenticated")
+	}
+
+	userID := claims.OwnerID()
+	user, err := s.admin.DB.FindUser(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	// user has not authorized github app
+	if user.GithubUsername == "" {
+		return nil, status.Error(codes.Unauthenticated, "not authenticated")
+	}
+
+	token, refreshToken, err := s.userAccessToken(ctx, user.GithubRefreshToken)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// refresh token changes after using it for getting a new token
+	// so saving the updated refresh token
+	_, err = s.admin.DB.UpdateUser(ctx, claims.OwnerID(), &database.UpdateUserOptions{
+		DisplayName:         user.DisplayName,
+		PhotoURL:            user.PhotoURL,
+		GithubUsername:      user.GithubUsername,
+		GithubRefreshToken:  refreshToken,
+		QuotaSingleuserOrgs: user.QuotaSingleuserOrgs,
+		PreferenceTimeZone:  user.PreferenceTimeZone,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to update user: %w", err)
+	}
+
+	client := github.NewTokenClient(ctx, token)
+
+	// use a client with user's token to get installations
+	installations, _, err := client.Apps.ListUserInstallations(ctx, nil)
```

Comment:
Is it a conscious decision that we don't want to fetch the next page ?

Reply:
Added pagination. We should do the same in other places as well in a future PR